### PR TITLE
feat(integrations)!: Tracecat registry

### PR DESCRIPTION
**Pull request by topher-lo**
Related to #403

This PR introduces Tracecat Registry, which migrates all actions and integrations into a standalone python package `tracecat_registry`. Existing UDF-based integrations have also been generalized into templated actions as part of this PR.

## Breaking changes
- Registry MUST import in a package NOT subpackage or individual module
- Renamed `analyze_{ioc}` into `search_{ioc}` to differentiate search (GET) to scans (POST, wait then GET report)
- All action names have been shortened to avoid verbosity e.g. `integrations.crowdstrike.list_crowdstrike_alerts` -> `integrations.crowdstrike.list_alerts`
- Okra: Renamed `okta_user_id` arg to `user_id`
- Jira: Renamed args `atlassian_domain` to `domain`, `issue_summary` to `summary`, `issue_priority` to `priority_id`, and `issue_type` to `issue_type_id` and dropped `optional_fields` in OOTB templated action (can be cloned then edited)
- All `search_malware_sample` actions: `file_hash` arg has been shortened to `hash`
- Sentinel One: Update actions now take list of IDs as opposed to single ID. Exact match and contains match filters for get agents is now split into two separate actions

## What changed
- Rewrite UDFs into:
  - Generic wrappers of Python SDKs (`boto3`, `falconpy`, `slack-sdk`)
  - Templated actions (either of SDK wrappers or `core.http_request`)
- Move integrations dependencies from `tracecat` to `tracecat_registry`
- QA: Add simple validation test of templated actions
- Removed `.parent.parent` pathing in tests. Just use relative path (from repo path as root).

## Note
List of breaking API changes will be published alongside the next 0.11 release docs.

### CODE REVIEW

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -55,6 +55,7 @@ services:
       TRACECAT__REMOTE_REGISTRY_URL: ${TRACECAT__REMOTE_REGISTRY_URL}
     volumes:
       - ./tracecat:/app/tracecat
+      - ./registry:/app/registry
```
Comment (daryllimyt) [inline docker-compose.dev.yml:58]: Add this for prod docker compose too

File: registry/tracecat_registry/integrations/__init__.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from . import boto3, falconpy, ldap3, pymongo, slack_sdk
+
+__version__ = "0.1.0"
```
Comment (daryllimyt) [inline registry/tracecat_registry/integrations/__init__.py]:
```suggestion
```

File: tests/unit/test_registry.py [hunk 3]
```diff
@@ -1,174 +1,99 @@
...
-        registry._register_udfs_from_module(test_module, visited_modules=set())
-        udf = registry.get("test.test_function")
-        udf.validate_args(num="${{ path.to.number }}")
-        udf.validate_args(num=1)
-        with pytest.raises(RegistryValidationError):
-            udf.validate_args(num="not a number")
-    finally:
-        # Clean up
-        del sys.modules["test_module"]
+    yield test_module
```
Comment (daryllimyt) [inline tests/unit/test_registry.py]: try finally so it always cleans up

File: tracecat/registry.py [hunk 4]
```diff
@@ -455,20 +448,19 @@ def _load_remote_udfs(self, remote_registry_url: str, module_name: str) -> None:
     def _load_template_actions(self) -> None:
```
Comment (daryllimyt) [inline tracecat/registry.py:449]: TODO: parametrize this

File: tracecat/registry.py [hunk 5]
```diff
@@ -502,6 +494,12 @@ def _load_template_actions(self) -> None:
                 is_template=True,
                 origin="base",
             )
+            file_paths.append(file_path)
```
Comment (daryllimyt) [inline tracecat/registry.py]: nit: count maybe?

File: tracecat/registry.py [hunk 6]
```diff
@@ -779,7 +782,7 @@ async def run(
         """
 
         context = base_context.copy() | {"inputs": args, "layers": {}}
-        logger.info("Running template action", action=self.definition.action)
+        logger.info("Running template action", action=self.definition.name)
```
Comment (daryllimyt) [inline tracecat/registry.py]: keep as action

**Review by daryllimyt — CHANGES_REQUESTED** [review]
🚀 lfg

File: tracecat/registry.py [hunk 7]
```diff
@@ -779,7 +786,7 @@ async def run(
...
```
Comment (topher-lo) [inline tracecat/registry.py]:
```suggestion
        logger.info("Running template action", action=self.definition.action)
```

File: tracecat/registry.py (hunk 4 above)
Comment (topher-lo) [reply tracecat/registry.py:449]: KK will do in next PR

**Review by daryllimyt — APPROVED** [review]
LGTM

**Pull request merged**